src/platform/datapath_epoll.c

```diff
@@ -1272,6 +1278,12 @@ CxPlatSocketContextInitialize(
 
     if (LocalAddress && LocalAddress->Ipv4.sin_port != 0) {
         CXPLAT_DBG_ASSERT(LocalAddress->Ipv4.sin_port == Binding->LocalAddress.Ipv4.sin_port);
+    } else if (RemoteAddress && LocalAddress && LocalAddress->Ipv4.sin_port == 0) {
+        //
+        // A client socket being assigned the same port as a remote socket causes issues later
+        // in the datapath and binding paths. Check to make sure this case was not given to us.
+        //
+        CXPLAT_DBG_ASSERT(Binding->LocalAddress.Ipv4.sin_port != RemoteAddress->Ipv4.sin_port);
     }
```

Comment:
Can we rework this so the `else if` is only evaluated in debug?

Reply:
The compiler should optimize this away, since it's an empty statement in release mode. I'll double check.

Reply:
Also the if is also only debug, so I'd just hide this whole thing behind debug. But the compiler should get rid of it all anyway.

Reply:
Personally, I'd prefer not to depend on that.